src/validator.cpp

```diff
@@ -1355,4 +1404,231 @@ void Validator::ValidatorImpl::checkUnitForCycles(const ModelPtr &model, const U
+        auto item = component->variable(i);
+        if (!item->id().empty()) {
+            info = "  - variable '" + item->name() + "' in component '" + component->name() + "'\n";
+            addIdMapItem(item->id(), info, idMap);
+        }
+        // Equivalent variables.
+        for (size_t e = 0; e < item->equivalentVariableCount(); ++e) {
+            auto equiv = item->equivalentVariable(e);
+            if (equiv != nullptr) {
+                auto equivParent = owningComponent(equiv);
+                if (equivParent != nullptr) {
+                    // Skipping half of the equivalences to avoid duplicate reporting.
+                    std::string s1 = item->name() + component->name();
+                    std::string s2 = equiv->name() + equivParent->name();
+                    std::string mappingId = Variable::equivalenceMappingId(item, equiv);
+                    // Variable mapping.
+                    if ((s1 < s2) && (!mappingId.empty())) {
+                        info = "  - variable equivalence between variable '" + item->name() + "' in component '" + component->name();
+                        info += "' and variable '" + equiv->name() + "' in component '" + equivParent->name() + "'\n";
+                        addIdMapItem(mappingId, info, idMap);
+                    }
+                    // Connections.
+                    auto connectionId = Variable::equivalenceConnectionId(item, equiv);
+                    std::string connection = component->name() < equivParent->name() ? component->name() + equivParent->name() : equivParent->name() + component->name();
+                    if ((s1 < s2) && (!connectionId.empty()) && (reportedConnections.count(connection) == 0)) {
+                        reportedConnections.insert(connection);
+                        info = "  - connection between components '" + component->name() + "' and '" + equivParent->name();
+                        info += "' because of variable equivalence between variables '" + item->name();
+                        info += "' and '" + equiv->name() + "'\n";
+                        addIdMapItem(connectionId, info, idMap);
+                    }
+                }
+            }
+        }
+    }
+
+    // Resets.
+    for (size_t i = 0; i < component->resetCount(); ++i) {
+        auto item = component->reset(i);
+        if (!item->id().empty()) {
+            info = "  - reset at index " + std::to_string(i) + " in component '" + component->name() + "'\n";
+            addIdMapItem(item->id(), info, idMap);
+        }
+        if (!item->testValueId().empty()) {
+            info = "  - test_value in reset at index " + std::to_string(i) + " in component '" + component->name() + "'\n";
+            addIdMapItem(item->testValueId(), info, idMap);
+        }
+        if (!item->testValueMathId().empty()) {
+            info = "  - MathML child of test_value in reset at index " + std::to_string(i) + " in component '" + component->name() + "'\n";
```

Comment:
so this only checks the `math` element child for an id, not the full MathML string?

Reply:
Ah, good point!  I'd forgotten about the resets ... will do that now.

Reply:
aa2e8d22